packages/smst-to-text/lib/toText.ts

```diff
@@ -0,0 +1,305 @@
+import {assertUnknownSectionNode} from '@suitest/smst';
+import {
+	ConditionNode,
+	InlineTextNode,
+	PropertiesNode,
+	TestLineNode,
+	TestLineResultNode,
+} from '@suitest/smst/types/unistTestLine';
+
+type RenderTextFunc = (node: ExtendedInlineNodes) => string;
+
+type ExtendedInlineNodes = InlineTextNode | {
+	type: TestLineResultStatus | SingleEntryStatus,
+	value: string,
+};
+
+const nl = '\n';
+const tab = '  ';
+const emptyString = '[EMPTY STRING]';
+const notDefined = '[NOT DEFINED]';
+
+const formatNotDefined = <T>(val?: T, formatter: (val: T) => string = String): string => {
+	if (typeof val === 'undefined' || val === null) {
+		return notDefined;
+	}
+
+	if (val as unknown as string === '') {
+		return emptyString;
+	}
+
+	return formatter(val);
+};
+
+const format = {
+	cancel: 	'\u001b[0m',
+	subject: 	'\u001b[32m',	// green
+	code: 		'\u001b[36m',	// cyan
+	input: 		'\u001b[4m',	// underscore
+	success: 	'\u001b[32m',	// green
+	fail: 		'\u001b[31m',	// red
+	fatal: 		'\u001b[31m',	// red
+	warning: 	'\u001b[33m',	// yellow
+	exit: 		'\u001b[34m',	// blue
+	excluded: 	'\u001b[34m',	// blue
+};
+
+const formatString = (text: string, type: string): string => {
+	if (type in format) {
+		return format[type as keyof typeof format] + text + format.cancel;
+	}
+
+	return text;
+};
+
+/**
+ * Render a single text node as plain text
+ */
+const renderPlainTextNode: RenderTextFunc = (node?: ExtendedInlineNodes): string => node ? node.value : '';
+
+/**
+ * Render a single text node with ANSI styling
+ */
+const renderFormattedTextNode: RenderTextFunc = (node: ExtendedInlineNodes): string =>
+	formatString(node?.value ?? '', node?.type);
```

Comment:
for what needed checks for null and undefined? node declared as required argument.

Reply:
There is a misalignment in JSX factory, we'll need to update it in following release